### PR TITLE
Skip all-false nullness array in ArrayBlockBuilder

### PR DESCRIPTION
**Pull request by findepi**

### CODE REVIEW

File: presto-spi/src/main/java/com/facebook/presto/spi/block/ArrayBlockBuilder.java [hunk 1]
```diff
@@ -224,6 +225,7 @@ private void entryAdded(boolean isNull)
         }
         offsets[positionCount + 1] = values.getPositionCount();
         valueIsNull[positionCount] = isNull;
+        hasNullValue |= isNull;
```
Comment (sopel39) [inline presto-spi/src/main/java/com/facebook/presto/spi/block/ArrayBlockBuilder.java]: What about not growing `valueIsNull`?

Comment (findepi) [reply presto-spi/src/main/java/com/facebook/presto/spi/block/ArrayBlockBuilder.java]: You mean to make `valueIsNull` field nullable in the builder? That would work too.
@dain do you have a preference?

**Comment by findepi** [comment]
@dain  should i wait for your review?

File: presto-spi/src/main/java/com/facebook/presto/spi/block/ArrayBlockBuilder.java (hunk 1 above)
Comment (dain) [reply presto-spi/src/main/java/com/facebook/presto/spi/block/ArrayBlockBuilder.java]: My preference is for consistency with with other block builders and it looks like they all use this style.

Comment (sopel39) [reply presto-spi/src/main/java/com/facebook/presto/spi/block/ArrayBlockBuilder.java]: Yet if there are not nulls, then we shouldn't allocate and grow `valueIsNull` at all.

Comment (dain) [reply presto-spi/src/main/java/com/facebook/presto/spi/block/ArrayBlockBuilder.java]: That seems like a good assumption, but I'd like to be more confident that it really matters before adding the complexity.

**Comment by findepi** [comment]
> This looks good, but we should update the calls to createArrayBlockInternal in AbstractArrayBlock in another PR.

@dain which PR did you mean?

**Comment by dain** [comment]
> > This looks good, but we should update the calls to createArrayBlockInternal in AbstractArrayBlock in another PR.
> 
> @dain which PR did you mean?

In some future PR that someone will write one day :)

**Pull request merged**